### PR TITLE
Fix events, fix, and clean up race cond in oauths

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

File: src/containers/sign-on/SignOnProvider.tsx [hunk 1]
```diff
@@ -448,7 +461,9 @@ export class SignOnProvider extends Component<SignOnProps, SignOnState> {
       validateHandle: this.props.validateHandle,
       onMetaMaskSignIn: this.onMetaMaskSignIn,
       recordTwitterStart: this.onRecordTwitterStart,
-      recordTwitterComplete: this.props.recordTwitterComplete
+      recordTwitterComplete: this.props.recordTwitterComplete,
+      recordInstagramStart: this.onRecordInstagramStart,
+      recordInstagramComplete: this.props.recordInstagramComplete
```
Comment (jowlee) [inline src/containers/sign-on/SignOnProvider.tsx:466]: I  don't think we need to pass this and `recordTwitterComplete` down

File: src/containers/sign-on/components/desktop/TwitterOverlay.js [hunk 2]
```diff
@@ -76,7 +86,7 @@ const TwitterOverlay = props => {
                     className={styles.instagramButton}
                     textClassName={styles.btnText}
                     iconClassName={styles.btnIcon}
-                    onClick={props.onClick}
+                    onClick={this.onClickInstagram}
```
Comment (jowlee) [inline src/containers/sign-on/components/desktop/TwitterOverlay.js:89]: I don't think you need the `this` here

File: src/containers/sign-on/components/desktop/TwitterOverlay.js [hunk 3]
```diff
@@ -88,7 +98,7 @@ const TwitterOverlay = props => {
                   textLabel={messages.twitterButton}
                   textClassName={styles.btnText}
                   iconClassName={styles.btnIcon}
-                  onClick={props.onClick}
+                  onClick={this.onClickTwitter}
```
Comment (jowlee) [inline src/containers/sign-on/components/desktop/TwitterOverlay.js:101]: same as above

File: src/containers/sign-on/components/mobile/TwitterOverlay.js [hunk 4]
```diff
@@ -78,15 +88,15 @@ const TwitterOverlay = props => {
...
+                      onClick={this.onClickInstagram}
                       onSuccess={props.onInstagramLogin}
                       onFailure={props.onFailure}
                       text={messages.instagramButton}
                     />
                   )}
                   <TwitterAuthButton
                     showIcon={false}
-                    onClick={props.onClick}
+                    onClick={this.onClickTwitter}
```
Comment (jowlee) [inline src/containers/sign-on/components/mobile/TwitterOverlay.js:99]: no this needed

**Review by jowlee — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-oauth-fixes

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-oauth-fixes

**Pull request merged**